pkg/tsdb/prometheus/models/query.go

```diff
@@ -158,13 +151,13 @@ func calculatePrometheusInterval(model *QueryModel, timeInterval string, query b
 		adjustedInterval = calculatedInterval.Value
 	}
 
-	if model.Interval == varRateInterval || model.Interval == varRateIntervalAlt {
+	if model.Interval != nil && *model.Interval == varRateInterval || *model.Interval == varRateIntervalAlt {
 		// Rate interval is final and is not affected by resolution
 		return calculateRateInterval(adjustedInterval, timeInterval, intervalCalculator), nil
 	} else {
-		intervalFactor := model.IntervalFactor
-		if intervalFactor == 0 {
-			intervalFactor = 1
+		var intervalFactor int64 = 1
+		if model.IntervalFactor != nil && *model.IntervalFactor != 0 {
+			intervalFactor = *model.IntervalFactor
```

Comment:
This is the nicest way that we came up with to prevent ugly nested if-else blocks :)